File: homework/hw3/python/tictactoe_search.py
from algorithm.alpha_beta_search import AlphaBetaSearch
from algorithm.general_game_search import GeneralGameSearch
from problem.tictactoe import TicTacToeState

if __name__ == "__main__":

    t = TicTacToeState()
    fgs = GeneralGameSearch(t)

    fgs.search()
    
    abs = AlphaBetaSearch(t)

    abs.search()

File: homework/hw3/c++/tictactoe_search.cpp
#include "algorithm/alpha_beta_search.hpp"
#include "algorithm/general_game_search.hpp"
#include "problem/tictactoe.hpp"

int main(){

    TicTacToeState t;
    
    GeneralGameSearch<decltype(t)> fgs(t);

    fgs.search();
    
    AlphaBetaSearch<decltype(t)> abs(t);

    abs.search();

    return 0;
}